source/common/router/config_impl.cc

```diff
@@ -383,6 +383,41 @@ RouteConstSharedPtr PathRouteEntryImpl::matches(const Http::HeaderMap& headers,
   return nullptr;
 }
 
+RegexRouteEntryImpl::RegexRouteEntryImpl(const VirtualHostImpl& vhost,
+                                         const envoy::api::v2::Route& route,
+                                         Runtime::Loader& loader)
+    : RouteEntryImplBase(vhost, route, loader),
+      regex_(std::regex{route.match().regex(), std::regex::optimize}) {}
+
+void RegexRouteEntryImpl::finalizeRequestHeaders(Http::HeaderMap& headers) const {
+  RouteEntryImplBase::finalizeRequestHeaders(headers);
+
+  const Http::HeaderString& path = headers.Path()->value();
+  const char* query_string_start = strchr(path.c_str(), '?');
```

Comment:
In general, prefer using std::string and other C++ style libraries rather than the C style versions.

Reply:
Header strings are not C++ strings for performance reasons, so this won't work without a copy. In this case strchr is fine IMO. You might be able to use a C++ algo though and not copy.

Reply:
Ah, yes, good point. Since you know the size from HeaderString you should be able to use `std::find` and the other algos that take iterator style arguments.